### PR TITLE
Fix: Add missing death modules to GLA Terrorists

### PR DESCRIPTION
**Pull request by xezon**
This change adds the missing death modules to GLA Terrorists. They now behave identical to other infantry units. The gameplay is unchanged.

* ~~Terrorists burn when burned and lasered~~
* Terrorists fling in air when burned and exploded
* Terrorists play crush sound when crushed and splatted

## Patched

Burned death is no longer representative.

https://github.com/TheSuperHackers/GeneralsGamePatch/assets/4720891/ba8094fc-864f-497e-83c8-41942cdec8ef

### CODE REVIEW

File: Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini [hunk 1]
```diff
@@ -1442,8 +1457,9 @@ Object GLAInfantryTerrorist
 
 
   ; I have just pulled my ripcord, and this ain't no parachute!
+  ; Patch104p @fix xezon 09/07/2023 Adds EXPLODED death type.
   Behavior = SlowDeathBehavior ModuleTag_Death03
-    DeathTypes          = NONE +SUICIDED
+    DeathTypes          = NONE +SUICIDED +EXPLODED
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: There is a discrepancy between this DeathBehavior module and the FireWeaponWhenDead module. EXPLOSION and SUICIDE will fling the guy into the air and do big damage. But when burned he does not fling into the air, despite doing big damage. Maybe we can make a FlamingInfantry version that is flung? That way it becomes even more obvious to the player/observer what happened.

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: Not sure if that will look good with the flaming infantry animation. It walks sideways.

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: Could add these animation states:

```ini
    TransitionState     = TRANS_Dying TRANS_Flailing
      Animation         = CIBURN_SKL.CIBURN_FLA
      AnimationMode     = ONCE
    End

    ConditionState      = DYING EXPLODED_FLAILING
      Animation         = CIBURN_SKL.CIBURN_FLA
      AnimationMode     = LOOP
      TransitionKey     = TRANS_Flailing
    End

    ConditionState      = DYING EXPLODED_BOUNCING
      Animation         = CIBURN_SKL.CIBURN_FLA
      AnimationMode     = ONCE
      TransitionKey     = None
    End
```
But it probably won't look right either.

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: I have added video of how it looks with this change. Burned death looks ok. The particle effect and sounds is clear.

Something is odd with the exploded death however. There is big sound and effect on top of Terrorist death. I have no yet figured where this comes from.

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: That is the RPG missiles blowing up by splash damage LOL

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: Ah ok. Now is clear. We already have report for it here: #1754

Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: If we can find a way to make flaming guy fling, we can add that later. For now this looks fine.

**Comment by xezon** [comment]
I think there is a problem with Flaming infantry: The Barrel is not a guy. That will look weird. So it is probably better to disable that part.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini (hunk 1 above)
Comment (xezon) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/GLAInfantry.ini]: Not necessary.

**Comment by xezon** [comment]
Flaming infantry death is now commented out and note is added why it does not fit.

Documentation added. Looks good.

**Pull request merged**